### DISCUSSION TITLE
Any thoughts on how steps can be integrated with Spring cloud stream

### DISCUSSION THREAD

**Discussion by c-ankur** [americanexpress/unify-flowret#15] [General] [answered]
Hi,
I would like to ask if there are any thoughts on how to integrate the step-based mechanism with Spring cloud stream so that the steps can be configured in a Functional programming model to submit to a Kafka topic, for example. This would make it much simpler to submit to topics in an event-driven architecture.

**Comment by deepakarora3** [comment, marked as answer]
I don't know much about Spring cloud stream but what I can say is that Flowret follows the imperative model of programming and not the reactive one. Having said this, there is nothing that stops anyone from implementing the steps in Java imperative model and submitting events to a Kafka queue.
